GameLogic/entities/MoveApplier.cs

```diff
@@ -24,6 +24,7 @@ public static class MoveApplier
         {
             var piece = (Piece)gameState.Board[move.StartCell];
             var deltaX = Math.Abs(move.EndCell.X - move.StartCell.X);
+            var lastPawnRanks = new List<int> { 0, 7 };
```

Comment:
I am wondering if we should generate `lastPawnRank` depending on the active colour.

Not critical, just an option.